### ISSUE TITLE
install steps Ubuntu

### ISSUE THREAD

**Issue by pierrextardif** [ASTex-ICube/real_time_glint#6]
When compiling on ubuntu 20.04, i get a linking error.

`cmake --version` -> `cmake version 3.20.1`
`glxinfo | grep "OpenGL version"` -> `OpenGL version string: 4.6.0 NVIDIA 465.19.01` .
And all dependencies seem fine: this is the result of my `cmake ..`  from `/real_time_glint/build` :

```sh

cmake ..
-- Running Linux compiling. Options [GLVND] for modern OpenGL or [LEGACY] for historical OpenGl.
-- Setting GLVND option.
CMake Warning (dev) at /usr/local/lib/cmake/assimp-5.0/assimpTargets.cmake:54 (if):
  if given arguments:

    "ON"

  An argument named "ON" appears in a conditional statement.  Policy CMP0012
  is not set: if() recognizes numbers and boolean constants.  Run "cmake
  --help-policy CMP0012" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.
Call Stack (most recent call first):
  /usr/local/lib/cmake/assimp-5.0/assimp-config.cmake:1 (include)
  CMakeLists.txt:21 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring done
-- Generating done
-- Build files have been written to: ~/real_time_glint-master`

```
Then `make` :
```
[...]
Linking CXX executable real_time_glint
/usr/bin/ld: ../opengl/libopengl.a(imgui_impl_opengl3.cpp.o): in function `ImGui_ImplOpenGL3_Init(char const*)':
imgui_impl_opengl3.cpp:(.text+0x2c): undefined reference to `glGetIntegerv'
/usr/bin/ld: imgui_impl_opengl3.cpp:(.text+0x3d): undefined reference to `glGetIntegerv'
/usr/bin/ld: imgui_impl_opengl3.cpp:(.text+0x142): undefined reference to `glGetIntegerv'
```
After gogleing the error amnd trying to integrate more recent imgui source files, I had the same exact error.
It seems like the makefile structure is missing something?
I double checked my glm install and glfw, with according `.cmake` files located at `usr/local/lib/cmake/glm/` and `usr/local/lib/cmake/glfw3/` .
[MakeFile Update] :
```
if(glm_FOUND)
    message(STATUS "glm has been found")
endif()

if(glfw3_FOUND)
    message(STATUS "glfw3 has been found")
endif()
```
prints in the console :
```sh
-- glm has been found
-- glfw3 has been found
```

 I would assume maybe the linking is wrong somewhere in the makefile?

**Comment by sylvainthery** [comment]
Sorry for the delay
Do you have mesa development package installed ?  _libglu1-mesa-dev_

**Comment by pierrextardif** [comment]
yes i do :
```sh
libglu1-mesa-dev is already the newest version (9.0.1-1build1).
```
Haven't looked at it for a little while tbh, but would love to make this work, :)}/.

**Comment by sylvainthery** [comment]
I do not understand, I can not reproduce your problem.
I have an Ubuntu 20.04 (not with exactly same version)
 - OpenGL version string: 4.6.0 NVIDIA 460.80
- I test with cmake version 3.16 & 3.21

On my system it compile and run without any problem.

There is one thing you can try. It seems that it should be better to change in _real\_time\_glint/CMakeLists.txt_ (line 27)
the ${OPENGL\_gl\_LIBRARY} into OpenGL::GL

**Comment by pierrextardif** [comment]
Hi not sure where my error wqas comoing from, did a quick re install on different machine, steps to install are working , but I get a segmentation fault now , :(.

I have tried to change the ` ${OPENGL_gl_LIBRARY}` into `OpenGL::GL` with no luck in `real_time_glint/CMakeLists.txt (line 27)` .

I can do `cmake ..` ( from a build folder), then `make` .
But then I have nothing on screen, and right away a segmentation fault:
```sh
-------------------------------------------------------------
GL Vendor    : NVIDIA Corporation
GL Renderer  : NVIDIA GeForce RTX 3080/PCIe/SSE2
GL Version   : 4.6.0 NVIDIA 470.57.02
GL Version   : 4.6
GLSL Version : 4.60 NVIDIA
MSAA samples : 0
MSAA buffers : 0
-------------------------------------------------------------
Segmentation fault
```

Any clues?
Will try in debug mode, with visual studio code later this week, breakpoints in VSCode are not working right now for some reasons, :(.

**Comment by xavierchermain** [comment]
Hi Pierre, 

I think you will find your problem with the debugger. You can also try our new implementation here: https://github.com/ASTex-ICube/aa_real_time_glint.

**Comment by pierrextardif** [comment]
Hi @xavierchermain,
Great work on this!
Thanks, the new repo has had the same issue.
However, upon running [this script](https://github.com/openframeworks/openFrameworks/blob/master/scripts/linux/ubuntu/install_dependencies.sh),  ands doing [this install](https://github.com/obsproject/obs-studio/wiki/Install-Instructions#debian-based-build-directions), it worked - unrealted work on other subjects HA!-.
I can't be sure 100% sure where the error was coming from as there was quite a few packages, so can't point to which package was tripping it.
If you start from a fresh install on ubuntu `20.04` I'm positive this will show up again.
Will close for now.
